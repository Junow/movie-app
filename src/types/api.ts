export interface NowPlaying {
  page: number
  results: Results[]
  dates: Dates
  total_pages: number
  totla_results: number
}

export interface Dates{
  maximum: string
  minimum: string
}

export interface Results {
  poster_path: string | null
  adult: boolean
  overview: string
  release_date: string
  genre_ids: number[]
  id: number
  original_title: string
  original_language: string
  title: string
  backdrop_path: string | null
  popularity: number
  vote_count: number
  video: boolean
  vote_average: number
}

export interface Detail {
  adult: boolean
  backdrop_path: string | null
  belongs_to_collection: null | Object
  budget: number
  genres: {
    id: number
    name: string
  }[]
  homapage: string | null
  id: number
  imdb_id: string | null
  original_language: string
  original_title: string
  overview: string | null
  popularity: number
  poster_path: string | null
  production_companies: {
    name: string
    id: number
    logo_path: string | null
    origin_country: string
  }[]
  production_countries: {
    iso_3166_1: string
    name: string
  }[]
  release_date: string
  revenue: number
  runtime: number | null
  spoken_languages: {
    iso_639_1: string
    name: string
  }[]
  status: string
  tagline: string | null
  title: string
  video: boolean
  vote_average: number
  vote_count: number
  credits: {
    cast: Cast[]
    crew: Crew[]
  }
}

export interface Cast {
  cast_id: number
  character: string
  credit_id: string
  gender: number | null
  id: number
  name: string
  order: number
  profile_path: string | null
}

export interface Crew {
  credit_id: string
  department: string
  gender: number | null
  id: number
  job: string
  name: string
  profile_path: string | null
}