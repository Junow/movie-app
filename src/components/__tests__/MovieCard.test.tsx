import React from 'react'
import {render, RenderResult, fireEvent} from '@testing-library/react'
import NowPlaying, {MovieCardProps} from '../MovieCard'
import {Results} from '../../types/api'

describe('<NowPlaying/>', () => {
  const setup = (props: Partial<MovieCardProps>={}) => {
    const initialProps: MovieCardProps = {
      movie: {
        popularity: 346.971,
      vote_count: 303,
      video: false,
      poster_path: "/ylQRi3edixhzUiubw7LQP1YhI6W.jpg",
      id: 577922,
      adult: false,
      backdrop_path: "/wzJRB4MKi3yK138bJyuL9nx47y6.jpg",
      original_language: "en",
      original_title: "Tenet",
      genre_ids: [
          28,
          53
      ],
      title: "테넷",
      vote_average: 7.6,
      overview: "시간의 흐름을 뒤집는 인버전을 통해 현재와 미래를 오가며 세상을 파괴하려는 사토르를 막기 위해 투입된 작전의 주도자. 인버전에 대한 정보를 가진 닐과 미술품 감정사이자 사토르에 대한 복수심이 가득한 그의 아내 캣과 협력해 미래의 공격에 맞서 제3차 세계대전을 막아야 한다!",
      release_date: "2020-08-22"
      },
      handleClick: jest.fn()
    }

    return render(<NowPlaying {...initialProps} {...props}/>)
  }

  it('should render correctly', () => {
    const wrapper = setup()
    expect(wrapper).toMatchSnapshot()
  })

  it('should have correct title', () => {
    const {getByTestId} = setup()
    const title = getByTestId('title')
    expect(title).toHaveTextContent('테넷')
  })

  it('should fire function', () => {
    const handleClick = jest.fn()
    const {getByTestId} = setup({
      handleClick
    })
    const title = getByTestId('title')
    expect(title).toBeTruthy()
    if(!title) return
    fireEvent.click(title)
    expect(handleClick.mock.calls.length).toBe(1)
  })


})