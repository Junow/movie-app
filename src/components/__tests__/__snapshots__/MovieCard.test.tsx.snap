// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NowPlaying/> should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxirZ gatXeD"
      >
        <div
          class="sc-AxjAm cIJhUp"
        >
          <div
            class="lazy-image"
          />
          <img
            class="image"
            data-src="https://image.tmdb.org/t/p/w500/ylQRi3edixhzUiubw7LQP1YhI6W.jpg"
          />
        </div>
        <div>
          <h3
            data-testid="title"
          >
            테넷
          </h3>
          <span>
            2020-08-22
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxirZ gatXeD"
    >
      <div
        class="sc-AxjAm cIJhUp"
      >
        <div
          class="lazy-image"
        />
        <img
          class="image"
          data-src="https://image.tmdb.org/t/p/w500/ylQRi3edixhzUiubw7LQP1YhI6W.jpg"
        />
      </div>
      <div>
        <h3
          data-testid="title"
        >
          테넷
        </h3>
        <span>
          2020-08-22
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
